### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by PavelGozhii**

### CODE REVIEW

File: Main.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package com.company;
+
+import com.company.hw01.*;
+
+import java.lang.reflect.Array;
+
+public class Main {
+
+    public static void main(String[] args) {
+	// write your code here
```
Comment (Siril-O) [inline Main.java:10]: Remove this commented code

File: Matrix.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import java.io.Serializable;
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class Matrix  extends Object implements Serializable {
+    private int[][] matrix;
+    private int height;
+    private int width;
+    private Matrix existring = null;
+    private boolean sorted = false;
```
Comment (Siril-O) [inline Matrix.java:12]: false can be removed

File: Matrix.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+import java.io.Serializable;
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class Matrix  extends Object implements Serializable {
+    private int[][] matrix;
+    private int height;
+    private int width;
+    private Matrix existring = null;
```
Comment (Siril-O) [inline Matrix.java:11]: null can be removed

File: Matrix.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        this.width = width;
+    }
+
+    public Matrix(int[][] matrix){
+        existring = this;
+        this.matrix = matrix;
+    }
+
+    public void printMatrix() {
+        if (existring != null) {
```
Comment (Siril-O) [inline Matrix.java:26]: Why do you need existring , you dont use it in calculations?

Comment (Siril-O) [reply Matrix.java:26]: Remove it or explain what is the purpose to check is it is not null. Also existring  should be existing

File: Matrix.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        if(existring != null){
+            if(matrix.length != matrix[0].length){
+                System.out.println("Matrix is not square");
+                return null;
+            }else{
+                int[] array = new int[matrix.length];
+                for(int i = 0; i < matrix.length; i++){
+                    array[i] = matrix[i][i];
+                }
+                for(int i = 0; i < array.length; i++){System.out.print(array[i]);}
```
Comment (Siril-O) [inline Matrix.java:46]: Wrong format lock should be on new line

File: Object.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.company.hw01;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+public class Object {
```
Comment (Siril-O) [inline Object.java:8]: why it is called Object ?

File: Task1.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.company.hw01;
+
+public class Task1 {
+    public int resoltOfMatch(int firstResult, int secondResult, int firtsPromise, int secondPromise){
```
Comment (Siril-O) [inline Task1.java:4]: typo in firtsPromise

File: Task1.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.company.hw01;
+
+public class Task1 {
+    public int resoltOfMatch(int firstResult, int secondResult, int firtsPromise, int secondPromise){
+        return (firstResult == firtsPromise) && (secondResult == secondPromise) ? 2 :
+                (firstResult > secondResult && firtsPromise > secondPromise)
+                        || (firstResult < secondResult && firtsPromise < secondPromise)
+                        || (firstResult == secondResult && firtsPromise == secondPromise) ? 1 : 0;
```
Comment (Siril-O) [inline Task1.java:8]: you can avoid (firstResult == secondResult && firtsPromise == secondPromise) by changing previous line

File: Task3.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
+package com.company.hw01;
+
+public class Task3 {
+    public double getPow(float x, double pow){
```
Comment (Siril-O) [inline Task3.java:4]: Bad name, you are not getting Pow you are calculating it

File: Task3.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+        if(pow == 0){
+            return 1;
+        }
+        for(int i = 0; i < pow; i ++){
+            result*=x;
+        }
+        if(pow > 0){
+            return result;
+        }else if(pow < 0){
+            return 1/result;
```
Comment (Siril-O) [inline Task3.java:17]: 1/result   does not calculates negative power

File: Task3.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
+package com.company.hw01;
+
+public class Task3 {
+    public double getPow(float x, double pow){
+        int m = 1;
+        double result = x;
+        int exp = 0;
+        if(pow == 0){
```
Comment (Siril-O) [inline Task3.java:8]: Is this check necessary ?

File: Task3.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
+package com.company.hw01;
+
+public class Task3 {
+    public double getPow(float x, double pow){
+        int m = 1;
+        double result = x;
+        int exp = 0;
```
Comment (Siril-O) [inline Task3.java:7]: Why do you need m and exp variables ?

File: Task4.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
+package com.company.hw01;
+
+public class Task4 {
+    public long getFactorial(int n){
+        if(n == 1) return 1;
+        return  n * getFactorial(n-1);
```
Comment (Siril-O) [inline Task4.java:6]: Format spaces

File: Task4.java [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+        }
+        if(n == 1){
+            return 1;
+        }
+        else
+            return getFibonacci(n-1) + getFibonacci(n-2);
+    }
+
+    public float getCosTaylorSeries(float n, float x){
+        if(n > 999){
```
Comment (Siril-O) [inline Task4.java:21]: Why do you perform this check ?

File: Task5.java [hunk 15]
```diff
@@ -0,0 +1,85 @@
+package com.company.hw01;
+
+public class Task5 {
+
+    public int[] bubleSort(int[] arr) {
+        for (int i = arr.length; i > 0; i--) {
+            for (int e = 0; e < (i - 1); e++) {
+                if (arr[e] > arr[e + 1]) {
+                    arr[e] = arr[e] ^ arr[e + 1];
```
Comment (Siril-O) [inline Task5.java:9]: Create swap method and do not use ^, use some temporary variable

Comment (Siril-O) [reply Task5.java:9]: Remove duplicates of swap functionality in all methods

File: Task2.java [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+            for(int e = 0; e < escapes; e++){
+                System.out.print(" ");
+            }
+            stars = size - escapes;
+            for(int e = 0; e < stars; e++){
+                System.out.print("* ");
+            }
+            System.out.println();
+        }
+        for(int i = size-2; i >= 0; i--){
```
Comment (Siril-O) [inline Task2.java:19]: why size -2 ? Will this work to print 10  stars rhombus ?

File: Object.java [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+public class Object {
+    public static Object deepClone(Object object){
+        try {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            ObjectOutputStream outputStrm = new ObjectOutputStream(outputStream);
+            outputStrm.writeObject(object);
```
Comment (Siril-O) [inline Object.java:13]: Where it writes to and reads from ? And what task is it ?

Comment (Siril-O) [reply Object.java:13]: Please read  what should be done in 1.7  task

**Review by Siril-O — CHANGES_REQUESTED** [review]

File: Matrix.java (hunk 4 above)
Comment (PavelGozhii) [reply Matrix.java:26]: > Remove it or explain what is the purpose to check is it is not null. Also existring should be existing

нужно проверить на заполнение данными матрицы, я ведь не хочу, что бы операции были из пусто матрицей

File: Task2.java (hunk 16 above)
Comment (PavelGozhii) [reply Task2.java:19]: > why size -2 ? Will this work to print 10 stars rhombus ?

поставил 10, работает, но все равно кое что исправил

**Pull request merged**